google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/AppEngineRuntimeInstance.java

```diff
@@ -61,32 +63,42 @@ public void deploy(@NotNull final DeploymentTask<AppEngineDeploymentConfiguratio
       return;
     }
 
-    AppEngineHelper appEngineHelper = new CloudSdkAppEngineHelper(task.getProject());
-
     AppEngineDeploymentConfiguration deploymentConfig = task.getConfiguration();
 
-    final CancellableRunnable deployRunner =  appEngineHelper.createDeployRunner(
-        logManager.getMainLoggingHandler(),
-        task.getSource(),
-        deploymentConfig,
-        callback);
+    AppEngineEnvironment environment =
+        AppEngineEnvironment.valueOf(deploymentConfig.getEnvironment());
+    if (environment == AppEngineEnvironment.APP_ENGINE_FLEX
+        && Strings.isNullOrEmpty(deploymentConfig.getYamlPath())) {
+      callback.errorOccurred(GctBundle.getString("appengine.deployment.error.staging.yaml")
+          + "\n" + GctBundle.getString("appengine.deployment.error.suggestflexfacet"));
```

Comment:
The only way for the YAML file to be missing at this stage is if it was swiped from under then run config's feet, which is what happens when the deployables suddenly go away, or the run config is otherwise reset. In my tests, I found that this was the only way when that happened.

I plan on removing this warning after the Flex facet has been live. It probably warrants adding a TODO here, though.